### ISSUE TITLE
[Bug]: 'Arithmetic Engine' ALU component keeps getting deleted

### ISSUE THREAD

**Issue by sinesc** [Turing-Complete-Game/Suggestions-and-Issues#134]
### What happened?

Premise: I have previously completed the Arithmetic Engine map but wanted to re-complete it for score sync (unrelated issue). My previous ALU is located in /Arch/Overture/ALU and the architecture that loads with the 'Add 5' map is using this component.

1) After completing 'Arithmetic Engine' the game creates a new /ALU component and replaces the previous ALU component used on the map 'Add 5' with it. (Expected: Don't modify the architecture, component footprint likely will be incompatible, e.g. the pins of my first ALU weren't even reachable and deleted wires)
2) After exiting the game and starting it again, the component used on the map 'Add 5' reverts back to the old /Arch/Overture/ALU. (Expected: Don't magically revert replacement (if replacement was actually intended))

3) Additionally, with Steam sync enabled the newly created /ALU/circuit.data is reproducibly deleted a few seconds after the game exits, leaving an empty ALU folder. I am unsure if this is a TC, Steam or combination of both bug. (Expected: Don't delete my ALU)

Will send game save sent via bug report feature.

Steps to reproduce:
- Launch game through Steam
- Start map 'Arithmetic engine', run Ticks
  Ticks complete, game loads component factory map (with wrong component, possibly a separate bug)
- Go to level map, start 'Add 5'
  The ALU component on that map is now the non-rectangular new /ALU
- Run ticks, go 'Onwards'
- Exit the game (ALU/ciruit.data is deleted a few seconds after)
- Start the game
- Start map 'Add 5'
  The ALU component on that map is now the old rectangular /Arch/Overture/ALU (regardless of Steam sync)
- Go to Component Factory -> Switch schematic
  (there should now be an empty 'ALU' folder, the ALU is gone [Steam sync only])

### Version

Latest

### Operating System

Windows 7

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by sinesc** [comment]
Update: I attempted to reproduce this on Linux. The circuit.data file was not deleted. Additionally, after this Steam on Windows asked whether I wanted to download from the cloud or upload my local save. I downloaded and now the issue is no longer reproducible on Windows either, even with the original game state as submitted through the bug report feature.

A member on discord asked to check the Steam cloud logs, but neither the .log, nor .previous.log go back far enough (I restarted Steam a couple times to test things).
